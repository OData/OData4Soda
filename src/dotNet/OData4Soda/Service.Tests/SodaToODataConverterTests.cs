using System;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using Socrata;

namespace Service.Tests
{
	[TestFixture, UseReporter(typeof (DiffReporter))]
	public class SodaToODataConverterTests
	{
		private static readonly DateTimeOffset FeedUpdateTime = DateTimeOffset.Parse("2011-12-03T15:50:26-08:00");
		private static readonly Uri RelativeUriOData = new Uri("/SomethingOData", UriKind.Relative);
		private static readonly Uri RelativeUriSoda = new Uri("/SomethingSoda", UriKind.Relative);

		[Test]
		public void FeedTest()
		{
			var testMessage = new InMemoryResponse();

			var converter = new SodaToODataConverter(testMessage, new Uri("http://fake"), TestData.TopLevelSodaResponse);
			var payload = new JsonPayload(TestData.SodaResponseFor(TestData.TopLevelSodaResponse));
			converter.ConvertFeed(RelativeUriOData, RelativeUriSoda, payload, FeedUpdateTime);

			ApproveResponse(testMessage);
		}

		[Test]
		public void FeedTestJson()
		{
			var testMessage = new InMemoryResponse();
			testMessage.SetHeader("Content-Type", "application/json");

			var converter = new SodaToODataConverter(testMessage, new Uri("http://fake"), TestData.TopLevelSodaResponse);
			var payload = new JsonPayload(TestData.SodaResponseFor(TestData.TopLevelSodaResponse));
			converter.ConvertFeed(RelativeUriOData, RelativeUriSoda, payload, FeedUpdateTime);

			ApproveResponse(testMessage);
		}

		[Test]
		public void MetadataTest()
		{
			var testMessage = new InMemoryResponse();

			var converter = new SodaToODataConverter(testMessage, new Uri("http://fake"), TestData.TopLevelSodaResponse);
			var payload = new JsonPayload(TestData.SodaResponseFor(TestData.TopLevelSodaResponse));
			converter.ConvertMetadata(RelativeUriOData, RelativeUriSoda, payload);

			ApproveResponse(testMessage);
		}

		private static void ApproveResponse(InMemoryResponse testMessage)
		{
			Approvals.Approve(Encoding.UTF8.GetString(testMessage.Stream.ToArray()) + "\r\n");
		}
	}
}