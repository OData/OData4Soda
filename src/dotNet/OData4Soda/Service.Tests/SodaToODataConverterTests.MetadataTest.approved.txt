<?xml version="1.0" encoding="utf-8"?>
<edmx:Edmx Version="3.0" xmlns:edmx="http://schemas.microsoft.com/ado/2009/11/edmx">
  <edmx:DataServices m:DataServiceVersion="3.0" xmlns:m="http://schemas.microsoft.com/ado/2007/08/dataservices/metadata">
    <Schema Namespace="OData4Socrata" xmlns="http://schemas.microsoft.com/ado/2009/11/edm">
      <ComplexType Name="url">
        <Property Name="url" Type="Edm.String" />
      </ComplexType>
      <ComplexType Name="phone">
        <Property Name="phone_number" Type="Edm.String" />
        <Property Name="phone_type" Type="Edm.String" />
      </ComplexType>
      <ComplexType Name="location">
        <Property Name="human_address" Type="Edm.String" />
        <Property Name="latitude" Type="Edm.String" />
        <Property Name="longitude" Type="Edm.String" />
        <Property Name="machine_address" Type="Edm.String" />
        <Property Name="needs_recoding" Type="Edm.Boolean" Nullable="false" />
      </ComplexType>
      <EntityType Name="Police_Stations">
        <Property Name="DISTRICT" Type="Edm.String" />
        <Property Name="ADDRESS" Type="Edm.String" />
        <Property Name="CITY" Type="Edm.String" />
        <Property Name="STATE" Type="Edm.String" />
        <Property Name="ZIP" Type="Edm.String" />
        <Property Name="WEBSITE" Type="OData4Socrata.url" Nullable="false" />
        <Property Name="PHONE" Type="OData4Socrata.phone" Nullable="false" />
        <Property Name="FAX" Type="Edm.String" />
        <Property Name="TTY" Type="Edm.String" />
        <Property Name="LOCATION" Type="OData4Socrata.location" Nullable="false" />
      </EntityType>
      <EntityContainer Name="">
        <EntitySet Name="Police_Stations" EntityType="OData4Socrata.Police_Stations" />
      </EntityContainer>
    </Schema>
  </edmx:DataServices>
</edmx:Edmx>
